e2etests/orchestration/upload_artifact_v1_test/main_test.go

```
@@ -26,15 +27,31 @@ const baseURL = "http://0.0.0.0:2080"
 
 func TestUploadArtifact(t *testing.T) {
 	srv := hoclient.NewHostOrchestratorService(baseURL)
-
-	if err := srv.UploadArtifact("../artifacts/cvd-host_package.tar.gz"); err != nil {
-		t.Fatal(err)
-	}
-
 	expected, err := hoclient.SHA256Checksum("../artifacts/cvd-host_package.tar.gz")
 	if err != nil {
 		t.Fatal(err)
 	}
+	_, err = srv.StatArtifact(expected)
+	if err == nil {
+		t.Fatal("expected error: file does not exist")
+	}
+
+	// Recreate two clients uploading the same artifact at the same time.
+	var wg sync.WaitGroup
+	for i := 0; i < 2; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			if err := srv.UploadArtifact("../artifacts/cvd-host_package.tar.gz"); err != nil {
+				t.Error(err)
+			}
+			if _, err := srv.StatArtifact(expected); err != nil {
+				t.Error(err)
+			}
```

IIUC, this test should fail if the implementation of `StatArtifact` is correct. If only a part of chunks are delivered, `StatArtifact` should return an error.

---

If the previous `UploadArtifact` call doesn't fail, it means the file was upload correctly, it means all chunks were properly persisted on the server side and it means that the checksum of the file on the server side should match the checksum of the original file, hence the StatArtifact call should not fail and the overall test should pass.